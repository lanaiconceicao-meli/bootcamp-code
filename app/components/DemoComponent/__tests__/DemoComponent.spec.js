/**
 * @jest-environment jsdom
 */
const React = require('react');
const { render, cleanup, fireEvent } = require('@testing-library/react');
const DemoComponent = require('..');

describe('DemoComponent', () => {
  afterEach(cleanup);

  const i18n = { gettext: text => text };

  test('should render correctly', () => {
    const { asFragment } = render(<DemoComponent i18n={i18n} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should increment counter', () => {
    const { getByText } = render(<DemoComponent i18n={i18n} />);

    fireEvent.click(getByText('+'));

    expect(getByText(/counter/)).toHaveTextContent('Clicks counter: 1');
  });

  test('should decrement counter', () => {
    const { getByText } = render(<DemoComponent i18n={i18n} />);

    fireEvent.click(getByText('-'));

    expect(getByText(/counter/)).toHaveTextContent('Clicks counter: -1');
  });
});
