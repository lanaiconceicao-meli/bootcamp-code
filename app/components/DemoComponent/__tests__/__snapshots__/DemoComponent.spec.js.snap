// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoComponent should render correctly 1`] = `
<DocumentFragment>
  <h1>
    This is a Demo component
  </h1>
  <p>
    <span
      class="demo-counter"
    >
      Clicks counter: 0
    </span>
    <button
      class="demo-button demo-button__add"
    >
      +
    </button>
    <button
      class="demo-button demo-button__subtract"
    >
      -
    </button>
  </p>
</DocumentFragment>
`;
